src/frontend/src/pages/AdminToolsPage/TeamsTools.tsx

```diff
@@ -131,18 +147,63 @@ const TeamsTools = () => {
               />
             </FormControl>
             <FormControl fullWidth>
-              <FormLabel>Description</FormLabel>
-              <ReactHookTextField
+              <Controller
                 name="description"
                 control={control}
-                fullWidth
-                multiline
-                rows={5}
-                errorMessage={errors.description}
-                maxLength={300}
+                rules={{ required: true }}
+                render={({ field }) => (
+                  <Box sx={{ display: 'flex-col' }}>
+                    <Box
+                      sx={{
+                        display: 'flex',
+                        alignItems: 'center',
+                        justifyContent: 'space-between',
+                        marginTop: '6px',
+                        marginBottom: '10px'
+                      }}
+                    >
+                      <FormLabel>Description</FormLabel>
+                      <Button
+                        onClick={() => {
+                          setIsPreview(!isPreview);
+                        }}
+                        sx={{
+                          ml: 2,
+                          backgroundColor: theme.palette.grey[600],
+                          color: theme.palette.getContrastText(theme.palette.grey[600]),
+                          '&:hover': {
+                            backgroundColor: theme.palette.grey[700]
+                          }
+                        }}
+                      >
+                        {isPreview ? 'Edit' : 'Preview'}
+                      </Button>
+                    </Box>
+                    {isPreview ? (
+                      <ReactMarkdown className={styles.markdown}>{currentDescription}</ReactMarkdown>
+                    ) : (
+                      <TextField
+                        fullWidth
+                        multiline
+                        rows={5}
+                        value={currentDescription}
+                        id={'description-input'}
+                        onChange={(e) => {
+                          setCurrentDescription(e.target.value);
+                          field.onChange(e);
+                        }}
+                        inputProps={{
+                          maxLength: isUnderWordCount(field.value, 300) ? null : 0
+                        }}
+                        error={!!errors.description || !isUnderWordCount(currentDescription, 300)}
```

Comment:
change all instances of currentDescription to field.value and remove all instances of setCurrentDescription